doc: clarify the behavior of `approx`

natmokval opened the pull request:
A small documentation fix to clarify the behavior of `approx`.

closes #13218

The-Compiler commented:
This seems like a very special case, so IMHO this should be mentioned somewhere further down with the other notes, but certainly not in the first two sentences before even explaining what the function is there for.

natmokval commented:
> This seems like a very special case, so IMHO this should be mentioned somewhere further down with the other notes, but certainly not in the first two sentences before even explaining what the function is there for.

Thanks @The-Compiler for your comment. I moved the note further down as you suggested and added an example. Could you please take a look at my update?

natmokval commented on src/_pytest/python_api.py:
```diff
@@ -632,6 +632,12 @@ def approx(expected, rel=None, abs=None, nan_ok: bool = False) -> ApproxBase:
         >>> ["foo", 1.0000005] == approx([None,1])
         False
 
+    Note that unlike built-in equality, this function considers
```
Thanks @nicoddemus for reviewing this PR!

I just pushed a commit to address your comment. CI is green.

nicoddemus commented on src/_pytest/python_api.py:
```diff
@@ -678,6 +678,12 @@ def approx(expected, rel=None, abs=None, nan_ok: bool = False) -> ApproxBase:
         from the
         `re_assert package <https://github.com/asottile/re-assert>`_.
 
+        Note that unlike built-in equality, this function considers
```
```suggestion

    .. note::
    
        Unlike built-in equality, this function considers
```

nicoddemus commented:
Thanks @natmokval!
